### ISSUE TITLE
Translate errors on website

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#433]

**Comment by cramerRF** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by strager** [comment]
When I created this task, I meant it for just the demo: https://quick-lint-js.com/demo/

**Comment by strager** [comment]
Any progress on this task? Do you need help?

I filed #664 for the documentation. I'll make this task (#433) for the web demo.

**Comment by barksanto** [comment]
Hey there I'm looking for OS tasks to assist with. Are you still looking for help with this one?

**Comment by strager** [comment]
> Are you still looking for help with this one?

Yes. I think @cramerRF isn't working on this task, so you can take it on.

If you want compensation (20 USD), please see https://quick-lint-js.com/hiring/

If you don't want compensation, and just want experience, I can hold your hand through this task.

**Comment by barksanto** [comment]
I'm really just looking for experience with this one. If you have any tips
on how I can get started I would really appreciate it.

**Comment by strager** [comment]
> If you have any tips on how I can get started I would really appreciate it.

Start by [running the website on your machine](https://github.com/quick-lint/quick-lint-js/tree/5c1666e466c44aeecfc618f8492f36def2f01616/website#developing). (Make sure to build quick-lint-js for WebAssembly so the web demo works.) Visit <http://localhost:9001/demo/> to see the demo. Hover over the red text to see the messages which should be translated.

There are four main parts:
* [JavaScript code responding to key presses](https://github.com/quick-lint/quick-lint-js/blob/5c1666e466c44aeecfc618f8492f36def2f01616/website/public/demo/demo.mjs)
* [JavaScript library setting up the WebAssembly module and providing a nicer API for to demo.mjs](https://github.com/quick-lint/quick-lint-js/blob/5c1666e466c44aeecfc618f8492f36def2f01616/website/wasm/quick-lint-js.js)
* [The c-api module which is directly called by the JavaScript library](https://github.com/quick-lint/quick-lint-js/blob/5c1666e466c44aeecfc618f8492f36def2f01616/src/c-api.cpp)
* quick-lint-js' main C++ code

I think we need to ultimately call the [`initialize_translations_from_locale` function](https://github.com/quick-lint/quick-lint-js/blob/5c1666e466c44aeecfc618f8492f36def2f01616/src/quick-lint-js/translation.h#L31). We could expose this through c-api in one of two ways:
1. A new function to set the locale
2. A new locale parameter for the existing `qljs_web_demo_lint` function

To start, I'd add `initialize_translations_from_locale("sv_SE");` at the beginning of `qljs_web_demo_create_document`. This should make all quick-lint-js messages Swedish (or at least the messages we have translations for). (If not, I'm misremembering something, or your dev setup is wrong.) From there, you should figure out how to make JavaScript control the locale.

**Comment by strager** [comment]
I added a `qljs_web_demo_set_locale` function in the C API. This is callable from JavaScript (`DocumentForWebDemo#setLocale`). I also added `qljs_list_locales` which is also callable from JavaScript (`Process#listLocales`).

Because all the C++/WebAssembly plumbing is done, this task should be much easier and exclusively front-end.

**Comment by strager** [comment]
@barksanto Are you still working on this task? Do you need any help?

**Comment by barksanto** [comment]
Hey Strager,

No, I'm not working on this one at the moment. Sorry I hope someone else
can help take over!

Thanks
